l1-contracts/contracts/state-transition/StateTransitionManager.sol

```diff
@@ -92,7 +92,7 @@ contract StateTransitionManager is IStateTransitionManager, ReentrancyGuard, Own
         chainAddresses = new address[](keys.length);
         uint256 keysLength = keys.length;
         for (uint256 i = 0; i < keysLength; ++i) {
-            chainAddresses[i] = hyperchainMap.get(i);
+            chainAddresses[i] = hyperchainMap.get(keys[i]);
```

Comment:
You can use branch `release-v23` instead of dev. There are latest changes including this fix